<?xml version="1.0"?>

<launch>

  <arg name="marker_size" doc="Size of the ArUco marker used, in meters" default="0.04"/>
  <arg name="marker_id" doc="The ID of the ArUco marker used" default="26" />

  <arg name="robot_name" default="px150"/>
  <arg name="robot_base_frame" default="$(arg robot_name)/base_link"/>

  <arg name="tracking_base_frame" value="camera_link" /> <!-- camera_bottom_screw_frame -->
  <arg name="marker_frame" value="tracking_marker" />

     <!-- start ArUco -->
  <node name="aruco_tracker" pkg="aruco_ros" type="single">
      <remap from="/camera_info" to="$(arg robot_name)/camera/color/camera_info"/> <!--  -->
      <remap from="/image" to="$(arg robot_name)/camera/color/image_raw"/> <!-- -->
      <param name="image_is_rectified" value="false"/>
      <param name="marker_size"        value="$(arg marker_size)"/>
      <param name="marker_id"          value="$(arg marker_id)"/>
      <param name="reference_frame"    value="$(arg tracking_base_frame)"/>
      <param name="camera_frame"       value="camera_color_optical_frame"/>
      <param name="marker_frame"       value="$(arg marker_frame)" />
  </node>
</launch>
